### PR TITLE
fix(k8s): fixed k8s scanner

### PR DESCRIPTION
**Pull request by drochow**
## Description
Fixed multiple issues with k8s scanner. 
Added posibility to filter component versions by component name. 

## What type of PR is this? (check all applicable)

- [ ] 🍕 Feature
- [x] 🐛 Bug Fix
- [ ] 📝 Documentation Update
- [ ] 🎨 Style
- [ ] 🧑‍💻 Code Refactor
- [ ] 🔥 Performance Improvements
- [ ] ✅ Test
- [ ] 🤖 Build
- [ ] 🔁 CI
- [ ] 📦 Chore (Release)
- [ ] ⏩ Revert

## Related Tickets & Documents

Related to issue #247 

## Added tests?

- [x] 👍 yes
- [ ] 🙅 no, because they aren't needed
- [ ] 🙋 no, because I need help
- [ ] Separate ticket for tests # (issue/pr)

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

## Added to documentation?

- [ ] 📜 README.md
- [ ] 🤝 Documentation pages updated
- [x] 🙅 no documentation needed
- [ ] (if applicable) generated OpenAPI docs for CRD changes

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: internal/database/mariadb/issue.go [hunk 1]
```diff
@@ -450,7 +450,11 @@ func (s *SqlDatabase) AddComponentVersionToIssue(issueId int64, componentVersion
 
 	_, err := performExec(s, query, args, l)
 
-	return err
+	if err != nil {
```
Comment (MR2011) [inline internal/database/mariadb/issue.go:453]: isn't that unnecessary?

File: internal/app/issue/issue_handler.go [hunk 2]
```diff
@@ -247,6 +248,9 @@ func (is *issueHandler) AddComponentVersionToIssue(issueId, componentVersionId i
 	err := is.database.AddComponentVersionToIssue(issueId, componentVersionId)
 
 	if err != nil {
+		if strings.HasPrefix(err.Error(), "Error 1062") {
```
Comment (MR2011) [inline internal/app/issue/issue_handler.go:251]: We should define some Database Errors that are returned by the database layer. Here we have now a dependency to the mariadb

File: internal/app/event/event_registry.go [hunk 3]
```diff
@@ -53,7 +53,7 @@ func (er *eventRegistry) PushEvent(event Event) {
 func NewEventRegistry(db database.Database) EventRegistry {
 	return &eventRegistry{
 		handlers: make(map[EventName][]EventHandler),
-		ch:       make(chan Event, 100),
+		ch:       make(chan Event, 1000),
```
Comment (MR2011) [inline internal/app/event/event_registry.go:56]: Do we want to have that in the config?

File: internal/app/component_version/component_version_handler.go [hunk 4]
```diff
@@ -117,6 +118,9 @@ func (cv *componentVersionHandler) CreateComponentVersion(componentVersion *enti
 	newComponent, err := cv.database.CreateComponentVersion(componentVersion)
 
 	if err != nil {
+		if strings.HasPrefix(err.Error(), "Error 1062") {
```
Comment (MR2011) [inline internal/app/component_version/component_version_handler.go:121]: See my other comment regarding errors from the db

File: scanner/k8s-assets/processor/processor.go [hunk 5]
```diff
@@ -73,33 +73,84 @@ func NewProcessor(cfg Config) *Processor {
...
+			return "", fmt.Errorf("failed to create SupportGroup %s: %w", serviceInfo.SupportGroup, err)
+		} else {
+			supportGroupId = createSupportGroupResp.CreateSupportGroup.Id
+		}
+	} else {
+		supportGroupId = _supportGroupId
 	}
 
-	createServiceResp, err := client.CreateService(ctx, *p.Client, createServiceInput)
+	_, _ = client.AddServiceToSupportGroup(ctx, *p.Client, serviceId, supportGroupId)
```
Comment (MR2011) [inline scanner/k8s-assets/processor/processor.go:127]: Shouldn't we check the err?

File: scanner/k8s-assets/processor/processor.go [hunk 6]
```diff
@@ -73,33 +73,84 @@ func NewProcessor(cfg Config) *Processor {
...
+	}
+	listSupportGroupsResp, err := client.ListSupportGroups(ctx, *p.Client, &listSupportGroupsFilter)
 	if err != nil {
-		return "", fmt.Errorf("failed to create Service %s: %w", serviceInfo.Name, err)
+		return "", fmt.Errorf("Couldn't list support groups")
+	}
+
+	// Return the first item
+	if listSupportGroupsResp.SupportGroups.TotalCount > 0 {
+		for _, sg := range listSupportGroupsResp.SupportGroups.Edges {
```
Comment (MR2011) [inline scanner/k8s-assets/processor/processor.go:145]: If we check the totalCount why not:
```
supportGroupId = listSupportGroupResp.SupportGroups.Edges[0].Node.Id
```

File: internal/api/graphql/graph/baseResolver/component_version.go [hunk 7]
```diff
@@ -81,17 +85,23 @@ func ComponentVersionBaseResolver(app app.Heureka, ctx context.Context, filter *
 		case model.ComponentNodeName:
 			componentId = []*int64{pid}
 		}
-	}
-
-	if filter == nil {
-		filter = &model.ComponentVersionFilter{}
+	} else {
+		componentId = lo.Map(filter.ComponentID, func(id *string, _ int) *int64 {
```
Comment (MR2011) [inline internal/api/graphql/graph/baseResolver/component_version.go:89]: Can we use this helper?
https://github.com/cloudoperators/heureka/blob/main/internal/util/slice.go#L41-L55

File: internal/app/issue/issue_handler.go (hunk 2 above)
Comment (drochow) [reply internal/app/issue/issue_handler.go:251]: I agree

File: internal/app/component_version/component_version_handler.go (hunk 4 above)
Comment (drochow) [reply internal/app/component_version/component_version_handler.go:121]: Agree

File: internal/database/mariadb/issue.go (hunk 1 above)
Comment (drochow) [reply internal/database/mariadb/issue.go:453]: I think i did that for debuging

**Review by MR2011 — DISMISSED** [review]

**Review by MR2011 — APPROVED** [review]

**Pull request merged**